packages/ui/routes/mailings/__partials/mailings-filters.vue

```diff
@@ -8,57 +11,59 @@ export default {
   },
   data() {
     return {
-      templates: [],
       templatesIsLoading: true,
-      templateIsError: false,
       pickerCreatedStart: false,
       pickerCreatedEnd: false,
       pickerUpdatedStart: false,
       pickerUpdatedEnd: false,
-      localFilters: {
-        name: '',
-        templates: [],
-        createdAtStart: '',
-        createdAtEnd: '',
-        updatedAtStart: '',
-        updatedAtEnd: '',
-        tags: [],
-      },
     };
   },
-
+  computed: {
+    ...mapState(FOLDER, ['filters']),
+    ...mapState(TEMPLATE, ['templates']),
+  },
   watch: {
     $route: 'reset',
-    localFilters: {
-      handler: function (updatedFilters) {
-        this.$emit('change', updatedFilters);
-      },
-      deep: true,
-    },
-  },
-
-  async mounted() {
-    const { $axios } = this;
-    try {
-      const { items } = await $axios.$get(apiRoutes.templates());
-      this.templates = items;
-    } catch (error) {
-      this.templateIsError = true;
-    }
-    this.templatesIsLoading = false;
   },
 
   methods: {
     reset() {
-      this.localFilters = {
+      this.$store.commit(`${FOLDER}/${SET_FILTERS}`, {
         name: '',
         templates: [],
         createdAtStart: '',
         createdAtEnd: '',
         updatedAtStart: '',
         updatedAtEnd: '',
         tags: [],
-      };
+      });
+    },
+    handleNameChange: _.debounce(function (value) {
+      this.commitFilterChangeToStore({ name: value });
+    }, 300),
+    handleTemplateChange(value) {
+      this.commitFilterChangeToStore({ templates: value });
+    },
+    handleTagsChange(value) {
+      this.commitFilterChangeToStore({ tags: value });
+    },
+    handleCreatedAtStartChange(value) {
+      this.commitFilterChangeToStore({ createdAtStart: value });
+    },
+    handleCreatedAtEndChange(value) {
+      this.commitFilterChangeToStore({ createdAtEnd: value });
+    },
+    handleUpdatedAtStartChange(value) {
+      this.commitFilterChangeToStore({ updatedAtStart: value });
+    },
+    handleUpdatedAtEndChange(value) {
+      this.commitFilterChangeToStore({ updatedAtEnd: value });
+    },
```

Comment:
can be refactored to something like this ?
```
handleFilterChange(filterKey, filterValue) {
  this.commitFilterChangeToStore({ [filterKey]: value });
}
```

And can be used later like this:
```
handleTemplateChange="handleFilterChange('createdAtEnd', $event)";
```